### PR TITLE
Transaction batching

### PR DESCRIPTION
**Pull request by SamueleA**
- [x] add transaction batching to existing purchase through sales contract
- [x] add currency abstraction option to the purchase modal

### CODE REVIEW

File: src/components/PurchaseModal/SwapAndBuy.tsx [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+       * @param tokenIds Token IDs to mint.
+       * @param amounts Amounts of tokens to mint.
+       * @param data Data to pass if receiver is contract.
+       * @param expectedPaymentToken ERC20 token address to accept payment in. address(0) indicates ETH.
+       * @param maxTotal Maximum amount of payment tokens.
+       * @param proof Merkle proof for allowlist minting.
+       * @notice Sale must be active for all tokens.
+       * @dev tokenIds must be sorted ascending without duplicates.
+       * @dev An empty proof is supplied when no proof is required.
+       */
```
Comment (ScreamingHawk) [inline src/components/PurchaseModal/SwapAndBuy.tsx]: Remove?

File: src/constants/index.ts [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+export const SALES_CONTRACT_ADDRESS = '0xe65b75eb7c58ffc0bf0e671d64d0e1c6cd0d3e5b'
+export const CHAIN_ID = 137
+export const ETHERSCAN_URL = 'https://polygonscan.com'
+export const TRANSACTION_CONFIRMATIONS = 4
+
+// Params for testing
+// export const NFT_TOKEN_ADDRESS = '0xe330d543e9189450c36dc873aa3ab14106b1ee87'
+// export const SALES_CONTRACT_ADDRESS = '0xfdd0d596350a78c3852a43d3b5910154b7c644db'
+// export const CHAIN_ID = 11155111
+// export const ETHERSCAN_URL = 'https://sepolia.etherscan.io'
```
Comment (ScreamingHawk) [inline src/constants/index.ts]: These could be env values? Keep these values as defaults when env value is not supplied

**Review by ScreamingHawk — DISMISSED** [review]
Approved. Couple nits

File: src/constants/index.ts (hunk 2 above)
Comment (SamueleA) [reply src/constants/index.ts]: Done in commit 165a036

File: src/components/PurchaseModal/SwapAndBuy.tsx (hunk 1 above)
Comment (SamueleA) [reply src/components/PurchaseModal/SwapAndBuy.tsx]: remove in commit [093807f](https://github.com/0xsequence/currency-abstraction-demo/pull/4/commits/093807f1eb0ffde78c2bd1a1cbfc8216d961db59)

**Review by ScreamingHawk — APPROVED** [review]
LGTM

**Pull request merged**